scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+import com.bittercode.model.Book;
+import com.bittercode.model.StoreException;
+import com.bittercode.service.BookService;
+import com.bittercode.util.DBUtil;
+
+public class BookServiceImpl implements BookService {
+
+    private static final String getAllBooksQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK;
+    private static final String getBookByIdQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+            + " WHERE " + BooksDBConstants.COLUMN_BARCODE + " = ?";
+
+    private static final String deleteBookByIdQuery = "DELETE FROM " + BooksDBConstants.TABLE_BOOK + "  WHERE "
+            + BooksDBConstants.COLUMN_BARCODE + "=?";
+
+    private static final String addBookQuery = "INSERT INTO " + BooksDBConstants.TABLE_BOOK + "  VALUES(?,?,?,?,?)";
+
+    private static final String updateBookQtyByIdQuery = "UPDATE " + BooksDBConstants.TABLE_BOOK + " SET "
+            + BooksDBConstants.COLUMN_QUANTITY + "=? WHERE " + BooksDBConstants.COLUMN_BARCODE
+            + "=?";
+
+    private static final String updateBookByIdQuery = "UPDATE " + BooksDBConstants.TABLE_BOOK + " SET "
+            + BooksDBConstants.COLUMN_NAME + "=? , "
+            + BooksDBConstants.COLUMN_AUTHOR + "=?, "
+            + BooksDBConstants.COLUMN_PRICE + "=?, "
+            + BooksDBConstants.COLUMN_QUANTITY + "=? "
+            + "  WHERE " + BooksDBConstants.COLUMN_BARCODE
+            + "=?";
+
+    @Override
+    public Book getBookById(String bookId) throws StoreException {
+        Book book = null;
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(getBookByIdQuery);
+            ps.setString(1, bookId);
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                String bCode = rs.getString(1);
+                String bName = rs.getString(2);
+                String bAuthor = rs.getString(3);
+                int bPrice = rs.getInt(4);
+                int bQty = rs.getInt(5);
+
+                book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+            }
+        } catch (SQLException e) {
+
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> getAllBooks() throws StoreException {
+        List<Book> books = new ArrayList<Book>();
+        Connection con = DBUtil.getConnection();
+
+        try {
+            PreparedStatement ps = con.prepareStatement(getAllBooksQuery);
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                String bCode = rs.getString(1);
+                String bName = rs.getString(2);
+                String bAuthor = rs.getString(3);
+                int bPrice = rs.getInt(4);
+                int bQty = rs.getInt(5);
+
+                Book book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+
+        }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Empty catch block in getAllBooks method</b></div>
<div id="fix">

The `catch` block for `SQLException` is empty. Consider logging the exception or handling it appropriately to avoid silent failures. A similar issue was also found in scr/main/java/bittercode/service/impl/BookServiceImpl.java (line 58-60).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -83,7 +83,7 @@
          } catch (SQLException e) {
 -
 +            throw new StoreException("Error retrieving books from database", e);
          }
          return books;
      }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/4#issuecomment-2434904657>#809440</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged